CMakeLists.txt

```diff
@@ -82,7 +82,7 @@ option(DISABLE_RCT2 "WIP: Try building without using code and data segments from
 option(USE_MMAP "Use mmap to try loading rct2's data segment into memory.")
 option(LAUNCHPAD_BUILD "Sets some build system options on launchpad.net")
 
-set(COMMON_COMPILE_OPTIONS "${COMMON_COMPILE_OPTIONS} -fstrict-aliasing -Werror -Wundef -Wmissing-declarations")
+set(COMMON_COMPILE_OPTIONS "${COMMON_COMPILE_OPTIONS} -fstrict-aliasing -Werror -Wundef -Wmissing-declarations -Winit-self -Warray-bounds -Waddress -Wchar-subscripts -Wenum-compare")
```

Comment:
Even though it doesn't get triggered for me, it may be worth checking.

See declaration of [`gResearchItems`](https://github.com/OpenRCT2/OpenRCT2/blob/cbf44741189ea85b5f50e7b6d861bdcfb750ff79/src/management/research.h#L78) which is getting passed to `move_research_item` in [`research_rides_setup`](https://github.com/OpenRCT2/OpenRCT2/blob/cbf44741189ea85b5f50e7b6d861bdcfb750ff79/src/windows/editor_inventions_list.c#L247) which in turn checks it with a "before" position, resulting in `gResearchItems - 1`. Perhaps worth double-checking.

This isn't strictly an error just yet, as I don't think there is anything wrong with simply checking against a pointer one before actual item. @IntelOrca anything you may want to add?